projects/igniteui-angular/src/lib/grids/grid/groupby-row.component.ts

```diff
@@ -8,26 +8,96 @@ import {
     Input,
     ViewChild,
     TemplateRef,
+    OnDestroy
 } from '@angular/core';
 import { IGroupByRecord } from '../../data-operations/groupby-record.interface';
 import { DataType } from '../../data-operations/data-util';
 import { GridBaseAPIService } from '../api.service';
 import { IgxGridBaseDirective } from '../grid-base.directive';
 import { IgxGridSelectionService, ISelectionNode } from '../selection/selection.service';
 import { GridType } from '../common/grid.interface';
+import { IgxFilteringService } from '../filtering/grid-filtering.service';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { GridSelectionMode } from '../common/enums';
 
 @Component({
     changeDetection: ChangeDetectionStrategy.OnPush,
     preserveWhitespaces: false,
     selector: 'igx-grid-groupby-row',
     templateUrl: './groupby-row.component.html'
 })
-export class IgxGridGroupByRowComponent {
+export class IgxGridGroupByRowComponent implements OnDestroy {
 
     constructor(public gridAPI: GridBaseAPIService<IgxGridBaseDirective & GridType>,
-        private gridSelection: IgxGridSelectionService,
+        public gridSelection: IgxGridSelectionService,
         public element: ElementRef,
-        public cdr: ChangeDetectorRef) { }
+        public cdr: ChangeDetectorRef,
+        public filteringService: IgxFilteringService) {
+            this.gridAPI.grid.onRowSelectionChange.pipe(takeUntil(this.destroy$)).subscribe(() => {
+                this.cdr.markForCheck();
+            });
+        }
+
+    /**
+     * @hidden
+     */
+    protected destroy$ = new Subject<any>();
+
+    /**
+     * @hidden @internal
+     */
+    public onGroupSelectorClick(event){
```

Comment:
These don't seem to follow formatting guidelines. Please use VS Code's format document option for these files